internal/pkg/policy/policy_output_integration_test.go

```diff
@@ -125,3 +127,74 @@ func TestRenderUpdatePainlessScript(t *testing.T) {
 		})
 	}
 }
+
+func TestPolicyOutputESPrepareRealES(t *testing.T) {
+	index, bulker := ftesting.SetupCleanIndex(context.Background(), t, dl.FleetAgents)
+
+	agentID := createAgent(t, index, bulker)
+	agent, err := dl.FindAgent(
+		context.Background(), bulker, dl.QueryAgentByID, dl.FieldID, agentID, dl.WithIndexName(index))
+	if err != nil {
+		require.NoError(t, err, "failed to find agent ID %q", agentID)
+	}
+
+	output := Output{
+		Type: OutputTypeElasticsearch,
+		Name: "test output",
+		Role: &RoleT{
+			Sha2: "new-hash",
+			Raw:  TestPayload,
+		},
+	}
+	policyMap := smap.Map{
+		"test output": map[string]interface{}{},
+	}
+
+	err = output.prepareElasticsearch(
+		context.Background(), zerolog.Nop(), bulker, &agent, policyMap)
+	require.NoError(t, err)
+
+	// need to wait a bit before querying the agent again
+	// TODO: find a better way to query the updated agent
+	time.Sleep(time.Second)
```

Comment:
You can use [require.Eventually](https://pkg.go.dev/github.com/stretchr/testify/require#Eventually) to poll for the state you need. There will still technically be a timeout, but it will allow the test to complete faster than the timeout by polling with shorter intervals.